PROVIDER_CHECKLIST.md

```diff
@@ -0,0 +1,48 @@
+# Provider Checklist
```

Comment:
Nit: Would you mind wrapping this at 80 cols? It's the convention for docs in this repo. https://marketplace.visualstudio.com/items?itemName=stkb.rewrap can automate it.